### PR TITLE
Copy index.lifecycle.name for ILM managed indices

### LINKED ISSUE #96732: The dowsample index has the ILM policy configured prematurely
### Elasticsearch Version

8.7+ 

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

Darwin

### Problem Description

The `downsample` action currently works as follows (high level details for a downsample action configured in the `warm` phase):

1. create a new index, the target index where the downsampled result will live. It has the name in the form “downsample-random-source_index_name” (this happens as part of the `rollup` ILM step)
2. copies the ILM execution step from the original index to the downsampled index (this will make the downsampled index appear as if it’s in the warm phase in ILM - same phase as the source index)
3. Replace the original index in the data stream with the downsampled index
4. Delete original index
5. Make the downsample index “look” like the original index (this is done by `SwapAliasesAndDeleteSourceIndexStep` - essentially transfer all aliases the original index had and the name of the original index, as aliases on the downsampled index ) 

Now, on step 1. we copy **most settings** the source index has on the target (`downsample-*`) index. We skip some though: https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleAction.java#L653

We do NOT currently skip `index.lifecycle.name`. This means that when the `downsample-*` index is created it'll have its own lifecycle policy and ILM will start executing it for it. 
This, at best, creates confusion as rollover will start failing for the `downsample-*` index due to not being bootstrapped for rollover. At worst, it could receive a policy that's been configured to mount the index as searchable snapshot, shrink, or other more disruptive action.

Logs will show things like:
```
downsample-ra12-.ds-metrics.testing-2023.06.09-000005) with ILM state from [null] to [{"phase":"new","action":"complete","name":"complete"}] in policy [metrics-ilm-policy]
...
downsample-ra12-.ds-metrics.testing-2023.06.09-000005) with ILM state from [{"phase":"hot","action":"rollover","name":"check-rollover-ready"}] to [{"phase":"hot","action":"rollover","name":"ERROR"}] in policy [metrics-ilm-policy]
	setting [index.lifecycle.rollover_alias] for index [downsample-ra12-.ds-metrics.testing-2023.06.09-000005] is empty or not defined
```

This proposes we fix this by creating the `downsample-*` index **without any** `index.lifecycle.name` setting configured and introducing another ILM step that will copy the `index.lifecycle.name` setting from the source index to the downsample index when we're ready for it to start its lifecycle - namely after step 2 in the high level description above (after `CopyExecutionStateStep`). Note we have a step already for this in ILM `CopySettingsStep` 

### Steps to Reproduce

Run downsample with ILM and check the logs for the `downsample-*` index 
https://www.elastic.co/guide/en/elasticsearch/reference/current/downsampling-ilm.html

### Logs (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by salvatore-campagna**
If an index is managed through ILM we copy the `index.lifecycle.name`
from the downsampling source index to the downsampling target index.
When creating the downsampling target index we make sure we do not
set the property so that the index is not picked by ILM soon after being
created. We set the lifecycle policy later, only when the index is ready for
ILM management. For this purpose we include a new step in the downsample
action whose purpose is just to copy the "index.lifecycle.policy" setting.

Note that the setting is not copied if the index is not managed. If a user
is running downsampling by means of the downsampling API it is up to the
user making sure that whatever ILM policy is applied later setting the
`index.lifecycle.name` once the target index is ready.

Resolves #96732

### CODE REVIEW

File: x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/downsample/TransportDownsampleActionTests.java [hunk 1]
```diff
@@ -98,6 +100,7 @@ private static void assertTargetSettings(final IndexMetadata indexMetadata, fina
             indexMetadata.getSettings().get(IndexMetadata.INDEX_BLOCKS_WRITE_SETTING.getKey()),
             settings.get(IndexMetadata.INDEX_BLOCKS_WRITE_SETTING.getKey())
         );
+        assertEquals(indexMetadata.getSettings().get(LifecycleSettings.LIFECYCLE_NAME), settings.get(LifecycleSettings.LIFECYCLE_NAME));
```
Comment (martijnvg) [inline x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/downsample/TransportDownsampleActionTests.java:103]: Maybe we should add this setting to `source` variable at line 25? Then we need to assert it isn't here? So check for null?

**Review by martijnvg — APPROVED** [review]
Left a small test comment, otherwise LGTM

File: x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/downsample/TransportDownsampleActionTests.java (hunk 1 above)
Comment (salvatore-campagna) [reply x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/downsample/TransportDownsampleActionTests.java:103]: We can't because it is "private"

**Review by andreidan — COMMENTED** [review]
Apologies @salvatore-campagna - I didn't get a notification to review this.

I had a quick look now and it LGTM. 

Thanks for fixing this.

**Pull request merged**